plugins/archiveviewer.koplugin/main.lua

```diff
@@ -252,8 +348,39 @@ function ArchiveViewer:viewFile(filepath)
 end
 
 function ArchiveViewer:extractFile(filepath)
-    local directory = util.splitFilePathName(self.arc.filepath)
+    local directory = self.extract_dir
+
+    if directory:sub(-1) ~= "/" then
+        directory = directory .. "/"
+    end
```

Comment:
`util.splitFilePathName` provides the trailing slash, but PathChooser (downloadmgr) doesn't.
You can just append it in `getChooseFolderButton()` and remove this check.